packages/ethereum/contracts/HoprChannels/Accounts.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.7.5;
+
+import "../utils/ECDSA.sol";
+
+contract Accounts {
+    /**
+     * @dev An account struct
+     */
+    struct Account {
+        bytes32 secret; // account's hashed secret
+        uint256 counter; // increases everytime 'secret' is changed
+    }
+
+    /**
+     * @dev Stored accounts keyed by their address
+     */
+    mapping(address => Account) public accounts;
+
+    /**
+     * @dev Initializes an account,
+     * stores it's public key, secret and counter,
+     * then emits {AccountInitialized} and {AccountSecretUpdated} events.
+     * @param account the address of the account
+     * @param pubKeyFirstHalf first half of the public key
+     * @param pubKeySecondHalf second half of the public key
+     * @param secret account's secret
+     */
+    function _initializeAccount(
+        address account,
+        uint256 pubKeyFirstHalf,
+        uint256 pubKeySecondHalf,
+        bytes32 secret
+    ) internal {
+        require(account != address(0), "account must not be empty");
+        require(pubKeyFirstHalf != uint256(0), "pubKeyFirstHalf must not be empty");
+        require(pubKeySecondHalf != uint256(0), "pubKeySecondHalf must not be empty");
+        // require(secret != bytes32(0), "secret must not be empty");
+
+        require(
+            ECDSA.pubKeyToEthereumAddress(pubKeyFirstHalf, pubKeySecondHalf) == account,
+            "public key does not match account"
+        );
+
+        _updateAccount(account, secret);
+
+        emit AccountInitialized(account, pubKeyFirstHalf, pubKeySecondHalf);
```

Comment:
try to emit 'AccountInitialized' before 'AccountSecretUpdated' so it's more clear

Reply:
dropping this in favour of including `secret` in `AccountInitialized`